test/e2e/suite_test.go

```diff
@@ -1736,3 +1738,66 @@ func checkInjectedContainersPresence(t *testing.T, ctx context.Context, f *frame
 	}
 	return nil
 }
+
+func checkInjectedVolumePresence(t *testing.T, ctx context.Context, f *framework.E2eFramework, dcKey framework.ClusterKey) error {
+	t.Logf("check that volumes were injected in %s cass pods in cluster %s", dcKey.Name, dcKey.K8sContext)
+	cassdc := &cassdcapi.CassandraDatacenter{}
+	err := f.Get(ctx, dcKey, cassdc)
+	if err != nil {
+		return err
+	}
+
+	require.Equal(t, 1, len(cassdc.Spec.StorageConfig.AdditionalVolumes), "expected 1 additional volume")
+	require.Equal(t, "/etc/extra", cassdc.Spec.StorageConfig.AdditionalVolumes[0].MountPath, "expected busybox-vol mount path")
+
+	_, found := cassandra.FindVolume(cassdc.Spec.PodTemplateSpec, "busybox-vol")
+	require.True(t, found, "busybox-vol volume not found in cassandra pod")
+
+	if containerIndex, containerFound := cassandra.FindContainer(cassdc.Spec.PodTemplateSpec, "busybox"); containerFound {
+		volumeMount := cassandra.FindVolumeMount(&cassdc.Spec.PodTemplateSpec.Spec.Containers[containerIndex], "busybox-vol")
+		require.NotNil(t, volumeMount, "busybox-vol volume mount not found in busybox container")
+		require.Equal(t, "/etc/busybox", volumeMount.MountPath, "expected busybox-vol mount path")
+	} else {
+		return fmt.Errorf("cannot find busybox injected container in pod template spec")
+	}
+
+	if initContainerIndex, initContainerFound := cassandra.FindInitContainer(cassdc.Spec.PodTemplateSpec, "init-busybox"); initContainerFound {
+		volumeMount := cassandra.FindVolumeMount(&cassdc.Spec.PodTemplateSpec.Spec.InitContainers[initContainerIndex], "busybox-vol")
+		require.NotNil(t, volumeMount, "busybox-vol volume mount not found in init-busybox container")
+		require.Equal(t, "/etc/busybox", volumeMount.MountPath, "expected busybox-vol mount path")
+	} else {
+		return fmt.Errorf("cannot find busybox injected container in pod template spec")
+	}
+
+	cassandraPods, err := getCassandraDatacenterPods(t, f, ctx, dcKey)
+	require.NoError(t, err, "failed listing Cassandra pods")
+	cassandraIndex, cassandraFound := findContainerInPod(t, cassandraPods[0], "cassandra")
+	require.True(t, cassandraFound, "cannot find cassandra container in cassandra pod")
+	volumeMount := cassandra.FindVolumeMount(&cassandraPods[0].Spec.Containers[cassandraIndex], "sts-extra-vol")
+	require.NotNil(t, volumeMount, "sts-extra-vol volume mount not found in cassandra container")
+	require.Equal(t, "/etc/extra", volumeMount.MountPath, "expected sts-extra-vol mount path")
+
+	return nil
+}
+
+func getCassandraDatacenterPods(t *testing.T, f *framework.E2eFramework, ctx context.Context, dcKey framework.ClusterKey) ([]corev1.Pod, error) {
```

Comment:
I might consider making this function a method of `framework.Framework` so it could be used in both integration and e2e tests. It's a minor suggestion though that doesn't need to hold up merging :)

Reply:
Good idea, I made the change.